### PR TITLE
Upgrade Kogito version

### PR DESCRIPTION
**Pull request by cristianonicolai**
**Check list**:

Your pull request:

- [x] targets the `development` branch
- [x] uses the `999-SNAPSHOT` version of Quarkus
- [x] has tests (`mvn clean test`)
- [x] works in native (`mvn clean package -Pnative`)
- [x] has native tests (`mvn clean verify -Pnative`)
- [x] makes sure the associated guide must not be updated
- [ ] links the guide update pull request (if needed)
- [ ] updates or creates the `README.md` file (with build and run instructions)
- [ ] for new quickstart, is located in the directory _component-quickstart_
- [ ] for new quickstart, is added to the root `pom.xml` and `README.md`

### CODE REVIEW

File: kogito-quickstart/pom.xml [hunk 1]
```diff
@@ -13,7 +13,7 @@
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
     <quarkus.platform.version>999-SNAPSHOT</quarkus.platform.version>
-    <kogito-quarkus.version>1.6.0.Final</kogito-quarkus.version>
+    <kogito-quarkus.version>1.8.0.Final</kogito-quarkus.version>
```
Comment (gsmet) [inline kogito-quickstart/pom.xml:16]: Why not 1.9.1.Final? That's the version used for the other quickstarts.

Comment (cristianonicolai) [reply kogito-quickstart/pom.xml:16]: there is a bug in that version, I'm working to fix it but most likely we will need a new release and then upgrade all.

**Review by gsmet — APPROVED** [review]
OK, thanks for the explanation!

**Comment by tarilabs** [comment]
sorry @cristianonicolai  I realized your extended tests here only now. I believe you rework this PR in order NOT to include `pom.xml` modifications, we can incorporate it. 
(I just manually checked your extended JUNit assertions locally and they pass as expected even when DRL reworked as a RuleUnit)

**Comment by cristianonicolai** [comment]
thanks @tarilabs, I've rebased the PR so it only includes the new tests. @gsmet should be ready to merge.

**Comment by cristianonicolai** [comment]
@gsmet anything else from my side that needs to be done in order to get this merged? thanks

**Comment by gsmet** [comment]
Sorry it fell through the cracks... I'm running CI and will merge once it's green.

**Comment by cristianonicolai** [comment]
@gsmet thanks

**Pull request merged**